### ISSUE TITLE
Compilation problem. Couldn't find referenced attribute error.

### ISSUE THREAD

**Issue by guptakeshav** [requery/requery#757]
Hey,
I have an entity defined something like this:

```
@Table(name = "Boards")
@Entity
interface IBoards : Persistable {
    @get:Column(nullable = false)
    @get:Key
    var boardId: String

    @get:Column(name = "gameId", nullable = false)
    @get:ManyToOne
    @get:Key
    var game: IGames

..... some other properties
}
```

and other like 
```
@Table(name = "Games")
@Entity
interface IGames : Persistable {
    @get:Column(nullable = false)
    @get:Key
    var id: String

..... some other properties
}
```

It used to compile perfectly when they were present in a single package. But I was refactoring them and hence moved them to different places. Now it shows me the following error:
```
com.sample.abc.models.Boards.gameId
w: 

w:     public abstract com.sample.xyz.models.IGames getGame();
w:                                                        ^
e: com/sample/abc/models/IBoards.java:20: error: Couldn't find referenced attribute null for com.sample.abc.models.Boards.gameId
e: 

e:     public abstract com.sample.xyz.models.IGames getGame();
```

**Comment by npurushe** [comment]
If you refactored from an IDE then it may not have modified the generated code, in which case a clean and rebuild would fix it

**Comment by guptakeshav** [comment]
I had tried clean and rebuild many times.

**Comment by guptakeshav** [comment]
There is a clear difference in the generated file of `IBoards` entity. 

a) When moved away from single package.
```
    public static final AttributeDelegate<Boards, IGames> GAME = new AttributeDelegate(
    new AttributeBuilder<Boards, IGames>("gameId", IGames.class)
    .setProperty(new Property<Boards, IGames>() {
        @Override
        public IGames get(Boards entity) {
            return entity.game;
        }

        @Override
        public void set(Boards entity, IGames value) {
            entity.game = value;
        }
    })
    .setPropertyName("getGame")
    .setPropertyState(new Property<Boards, PropertyState>() {
        @Override
        public PropertyState get(Boards entity) {
            return entity.$game_state;
        }

        @Override
        public void set(Boards entity, PropertyState value) {
            entity.$game_state = value;
        }
    })
    .setKey(true)
    .setGenerated(false)
    .setLazy(false)
    .setNullable(false)
    .setUnique(false)
    .setForeignKey(true)
    .setDeleteAction(ReferentialAction.CASCADE)
    .setUpdateAction(ReferentialAction.CASCADE)
    .setCascadeAction(CascadeAction.SAVE)
    .setCardinality(Cardinality.MANY_TO_ONE)
    .build());
```

b) When everything in same package.
```
public static final AttributeDelegate<Boards, IGames> GAME = new AttributeDelegate(
    new AttributeBuilder<Boards, IGames>("gameId", IGames.class)
    .setProperty(new Property<Boards, IGames>() {
        @Override
        public IGames get(Boards entity) {
            return entity.game;
        }

        @Override
        public void set(Boards entity, IGames value) {
            entity.game = value;
        }
    })
    .setPropertyName("getGame")
    .setPropertyState(new Property<Boards, PropertyState>() {
        @Override
        public PropertyState get(Boards entity) {
            return entity.$game_state;
        }

        @Override
        public void set(Boards entity, PropertyState value) {
            entity.$game_state = value;
        }
    })
    .setKey(true)
    .setGenerated(false)
    .setLazy(false)
    .setNullable(false)
    .setUnique(false)
    .setForeignKey(true)
    .setReferencedClass(Games.class)
    .setReferencedAttribute(new Supplier<Attribute>() {
        @Override
        public Attribute get() {
            return Games.ID;
        }
    })
    .setDeleteAction(ReferentialAction.CASCADE)
    .setUpdateAction(ReferentialAction.CASCADE)
    .setCascadeAction(CascadeAction.SAVE)
    .setCardinality(Cardinality.MANY_TO_ONE)
    .setMappedAttribute(new Supplier<Attribute>() {
        @Override
        public Attribute get() {
            return Games.BOARDS;
        }
    })
    .build());
```

**Comment by guptakeshav** [comment]
I am trying to move the classes to different modules, if that makes this issue?

**Comment by jbenezech** [comment]
I seem to run into that same issue trying to reference Entities defined in a android library from entities defined in an application

```
@ManyToOne()
UserProfile getUserProfile();
```
Where the holding entity is in the application and UserProfile is defined in a library project, will lead to that same error.

```
Error:(67, 17) error: Couldn't find referenced attribute null for AnnouncementEntity.userProfile
```

Adding

```
@ForeignKey(references = UserProfile.class)
```
Will change the error to 

```
Error:Couldn't find referenced attribute  for AnnouncementEntity.userProfile
```

In both cases, the generated Entity class does not have the ```setReferencedClass``` call which seems to cause this error.

More generally, I'm a bit lost on how to use entities from different android modules/librairies. A link to doc or sample would be greatly appreciated.

**Comment by jbenezech** [comment]
Here is a testcase that reproduces the issue: https://github.com/jbenezech/requery-issue-757

What seems to happen:
- The ```process``` method of the ```EntityProcessor``` is called twice. Once for the lib, once for the application. When called for the app, it receives only the annotations on the classes of the app.
- Requery computes the list of entities corresponding to these annotations and stores them in the ```EntityGraph```
- Because annotations from the lib are not passed in that round, entities from lib are not added to the graph
- When processing the ```@ManyToOne``` annotation, in ```referencingEntity``` of the ```EntityGraph```, only references to entities that are part of the graph are returned. Therefore the entity from lib is not.
- No reference is identified and the generated attribute lacks ```setReferencedClass``` and ```setReferencedAttribute```

As a side effect, lib entities not being part of the graph, they are not added to the generated Models.

So basically, this setup is not supported. Entities can only be related to entities from their own "annotation processing round".

The only workaround I found is to retrieve the generated Project.java, add the 2 missing methods, add this class to the project, remove its ```@Entity``` tag, clean to remove duplicates. And do that every time an app entity changes. Plus of course supply a custom Models, adding entities manually.
